symFileManager.py

```diff
@@ -48,38 +50,46 @@ def GetLibSymbolMap(self, libName, breakpadId):
     if libName == "":
       return None
 
+    # Guess the name of the .sym file on disk
+    if libName[-4:] == ".pdb":
+      symFileName = re.sub(r"\.[^\.]+$", ".sym", libName)
+    else:
+      symFileName = libName + ".sym"
+
+    pathSuffix = os.path.join(libName, breakpadId, symFileName)
+    urlSuffix = "/".join([libName, breakpadId, symFileName])
+
     # Check cache first
     libSymbolMap = None
     if libName in self.sCache and breakpadId in self.sCache[libName]:
       libSymbolMap = self.sCache[libName][breakpadId]
       self.UpdateMruList(libName, breakpadId)
+      self.diskCache.Use(pathSuffix)
```

Comment:
this GetLibSymbolMap function should be split into two functions, one for getting a libSymbolMap, the other for maintaining the in-memory cache